projects/ngx-fixture/src/lib/matchers.ts

```diff
@@ -0,0 +1,38 @@
+const elementMatchers: jasmine.CustomMatcherFactories = {
+
+  /**
+   * Checks that an element has the specified class
+   */
+  toHaveClass: (util: jasmine.MatchersUtil, customEqualityTesters: Array<jasmine.CustomEqualityTester>): jasmine.CustomMatcher => {
+    return {
+      compare: (el: any, expected: string): jasmine.CustomMatcherResult => {
+        if (!el) {
+          return { pass: false, message: `Expect element to have class '${expected}', but element was null or undefined` };
+        }
+        const actual = el.classes;
+        const pass = actual.includes(expected);
+        const message = `Expected element to have class '${expected}', but had '${actual.join(', ')}'`;
+        return { pass, message };
+      }
```

Comment:
Something's missing here, because all matchers should also support the case when they're used with `not`, and should return the appropriate error message. 
The simplest way to achieve that is probably to provide a compare() and a negativeCompare() function (see https://jasmine.github.io/2.0/custom_matcher.html#section-Custom_negative_comparators).

Reply:
Oh, and in the not case, I think the test should also fail if they pass null, or something other than a TestElement. Just as this test fails, for example, bacause toMatch is not supposed to be used on an object.

    const a = { hello: 'world' };
    expect(a).not.toMatch(/.*/);